src/App.js

```diff
@@ -0,0 +1,169 @@
+    // validation
+    if (!accountForm.userName) {
+      alert('usernmae must not be empty.');
+      return;
+    }
+    if (!/[a-z]+/.test(accountForm.password) || !/[A-Z]+/.test(accountForm.password) || accountForm.password.length < 8) {
+      alert('wrong format password.');
+      return;
+    }
+    if (accountForm.repeatPassword !== accountForm.password) {
+      alert('repeated password must be the same as the password');
+      return;
+    }
+    if (!/^1[3,5,6,8]\d{9}$/.test(accountForm.phoneNumber)) {
+      alert('wrong format phone number.');
+      return;
+    }
+    this.props.accountFormSubmit({
+      userName: accountForm.userName,
+      password: accountForm.password,
+      phoneNumber: accountForm.phoneNumber
+    });
+    this.setState({showJSON: true});
+  }
+
+  render() {
+    return (
+      <div className="App">
+        <Form
+          onSubmit={(event) => this.onSubmit(event)}
+        >
+          <FormGroup>
+            <Label for="userName">User Name</Label>
+            <Input 
+              type="text" 
+              name="userName" 
+              id="userName" 
+              placeholder="please input your name"
+              onInput={(event) => this.onFieldChange('userName', event.target.value)}
+            />
+            <FormText color="muted">
+              can not be empty.
+            </FormText>
+          </FormGroup>
+          <FormGroup>
+            <Label for="password">Password</Label>
+            <Input 
+              type="password" 
+              name="password" 
+              id="password" 
+              placeholder="please input your password"
+              onInput={(event) => this.onFieldChange('password', event.target.value)}
+            />
+            <FormText color="muted">
+              must be at least 8 characters, contain upper and lower case letters.
+            </FormText>
+          </FormGroup>
+          <FormGroup>
+            <Label for="repeatPassword">Repeat Password</Label>
+            <Input 
+              type="password" 
+              name="repeatPassword" 
+              id="repeatPassword" 
+              placeholder="please input your password again" 
+              onBlur={(event) => this.onRepeatPasswordFieldBlur(event.target.value)}
+            />
+          </FormGroup>
+          <FormGroup>
+            <Label for="phoneNumber">PhoneNumber</Label>
+            <InputGroup>
+              <InputGroupAddon>+86</InputGroupAddon>
+              <Input 
+                type="tel" 
+                name="phoneNumber" 
+                id="phoneNumber" 
+                placeholder="please input your phone number"
+                onInput={(event) => this.onFieldChange('phoneNumber', event.target.value)}
+              />
+            </InputGroup>
+            <FormText color="muted">
+              must be numbers and length equal to 11.
+            </FormText>
+          </FormGroup>
+          <Button 
```

Comment:
from the wiki, the button should be disabled first, only all conditions passed, enable it

Reply:
create a variable named btnDisable to listen whether or not the form validation passed